locations/spiders/allerganbeauty_jp.py

```diff
@@ -0,0 +1,75 @@
+from typing import Any
+
+from scrapy import Spider
+from scrapy.http import Response
+
+from locations.categories import Categories, PaymentMethods, apply_category, apply_yes_no
+from locations.dict_parser import DictParser
+
+
+class AllerganbeautyJPSpider(Spider):
+    name = "allerganbeauty_jp"
+
+    start_urls = ["https://clinics.allerganbeauty.jp/api/all_point"]
+    allowed_domains = ["clinics.allerganbeauty.jp"]
+    country_code = "JP"
+
+    TYPES = {
+        "皮": "dermatology",
+        "アレルギー": "allergology",
+        "婦人": "gynaecology",
+        "形成": "plastic_surgery",
+        "泌尿": "urology",
+        "内科": "internal",
+        "眼科": "ophthalmology",
+        "麻酔": "anaesthetics",
+        "耳鼻咽喉": "otolaryngology",
+        "口膣外科": "stomatology",
+        "漢方": "traditional_chinese_medicine",
+        "リウマチ": "rheumatology",
+        "消化器": "gastroenterology",
+        "小児": "paediatrics",
+        "リハビリテーション": "physiatry",
+        "腎臓": "nephrology",
+        "人間ドック": "general",
+        "美容歯": "implantology",
+        "肛門": "proctology",
+    }
+
+    def parse(self, response: Response, **kwargs: Any) -> Any:
+        for store in response.json()["shop_list"]:
+
+            item = DictParser.parse(store)
+            item["ref"] = store["key"]
+            try:
+                item["website"] = store["施設URL"]
+            except:
+                pass
+            try:
```

Comment:
Instead of using a `try/except` to catch the `KeyError`, use the `get()` function:


```suggestion
            item["website"] = store.get("施設URL")
```

Reply:
> The try/except are still in there.

The store.get function didn't work for some of the stores, since it would fail if they didn't have a website.

Reply:
> The try/except are still in there.

I have updated it, the phone number works fine but I still have to use the try except with the specialty function.